src/repo/artifacts/docker.go

```diff
@@ -31,11 +45,41 @@ func NewDockerArtifactory(registry string) (*DockerArtifact, error) {
 	}
 
 	return &DockerArtifact{
-		registry: registry,
-		store:    buildStore,
+		registry:          registry,
+		store:             buildStore,
+		registryTLSVerify: registryTLSVerify,
+		registryCertDir:   registryCertDir,
+		registryAuthType:  registryAuthType,
+		registryUsername:  registryUsername,
+		registryPassword:  registryPassword,
+		registryToken:     registryToken,
 	}, nil
 }
 
+func (a *DockerArtifact) getAuth() *types.DockerAuthConfig {
+	switch a.registryAuthType {
+	case "basic":
+		return &types.DockerAuthConfig{
+			Username: a.registryUsername,
+			Password: a.registryPassword,
+		}
+	case "token":
+		return &types.DockerAuthConfig{
+			IdentityToken: a.registryToken,
+		}
+	default:
```

Comment:
Better to have case "": return nil, and default: return domain.ErrUnknownDockerAuthType